### PR TITLE
IR-390: Make a configmap for MCO to consume CAs

### PR DESCRIPTION
**Pull request by cdoern**
Took Flavian's work on the current PR, and fixed the configmap already exists bug. This should be good to go in tandem with:  https://github.com/openshift/machine-config-operator/pull/3770

### CODE REVIEW

**Comment by cdoern** [comment]
it seems like the generator thinks the configmap both exists and doesn't exist....

**Comment by cdoern** [comment]
:crossed_fingers:

File: pkg/resource/imageregistryca.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+			Name:      girca.GetName(),
+			Namespace: girca.GetNamespace(),
+		},
+		Data:       map[string]string{},
+		BinaryData: map[string][]byte{},
+	}
+
+	var ownHostnameKeys []string
+
+	serviceCA, err := girca.lister.ConfigMaps(defaults.OpenShiftConfigManagedNamespace).Get(defaults.ServiceCAName)
```
Comment (dmage) [inline pkg/resource/imageregistryca.go:111]: `serviceca` is managed by the image-registry operator and its namespace is `openshift-image-registry`.

The generator is https://github.com/openshift/cluster-image-registry-operator/blob/master/pkg/resource/serviceca.go

**Comment by xiuwang** [comment]
`image-registry-ca` cm in openshift-config-managed keeps same content with `image-registry-certificates` cm of openshift-image-registry namespace.
Will check hypershift situation, and update here

**Comment by xiuwang** [comment]
Hypershift hosted cluster install failed with such error, will try more times
```
07-20 16:49:17.842  + aws s3api delete-public-access-block --bucket aos-hypershift-ci-oidc-5811
07-20 16:49:18.399  /usr/local/lib/python3.6/site-packages/OpenSSL/_util.py:6: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography. The next release of cryptography will remove support for Python 3.6.
07-20 16:49:18.399    from cryptography.hazmat.bindings.openssl.binding import Binding
07-20 16:49:18.399  
07-20 16:49:18.399  An error occurred (NoSuchBucket) when calling the DeletePublicAccessBlock operation: The specified bucket does not exist
07-20 16:49:18.399  + export hypershift_bucket_name=aos-hypershift-ci-oidc-5811
07-20 16:49:18.399  + hypershift_bucket_name=aos-hypershift-ci-oidc-5811
07-20 16:49:18.399  + echo '{
07-20 16:49:18.399              "Version": "2012-10-17",
07-20 16:49:18.399              "Statement": [
07-20 16:49:18.399                  {
07-20 16:49:18.399                  "Effect": "Allow",
07-20 16:49:18.399                  "Principal": "*",
07-20 16:49:18.399                  "Action": "s3:GetObject",
07-20 16:49:18.399                  "Resource": "arn:aws:s3:::${hypershift_bucket_name}/*"
07-20 16:49:18.399                  }
07-20 16:49:18.399              ]
07-20 16:49:18.399          }'
07-20 16:49:18.399  + envsubst
07-20 16:49:18.399  + cat policy.json
07-20 16:49:18.399  {
07-20 16:49:18.399              "Version": "2012-10-17",
07-20 16:49:18.399              "Statement": [
07-20 16:49:18.399                  {
07-20 16:49:18.399                  "Effect": "Allow",
07-20 16:49:18.399                  "Principal": "*",
07-20 16:49:18.399                  "Action": "s3:GetObject",
07-20 16:49:18.399                  "Resource": "arn:aws:s3:::aos-hypershift-ci-oidc-5811/*"
07-20 16:49:18.399                  }
07-20 16:49:18.399              ]
07-20 16:49:18.399          }
07-20 16:49:18.399  + aws s3api put-bucket-policy --bucket aos-hypershift-ci-oidc-5811 --policy file://policy.json/
07-20 16:49:18.958  /usr/local/lib/python3.6/site-packages/OpenSSL/_util.py:6: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography. The next release of cryptography will remove support for Python 3.6.
07-20 16:49:18.958    from cryptography.hazmat.bindings.openssl.binding import Binding
07-20 16:49:18.958  
07-20 16:49:18.958  An error occurred (NoSuchBucket) when calling the PutBucketPolicy operation: The specified bucket does not exist
07-20 16:49:18.958  + return 255
07-20 16:49:18.958  + '[' 255 -eq 0 ']'
07-20 16:49:18.958  + echo 'try 1 time to create s3 bucket aos-hypershift-ci-oidc-5811'
07-20 16:49:18.958  try 1 time to create s3 bucket aos-hypershift-ci-oidc-5811
07-20 16:49:18.958  + n=2
07-20 16:49:18.958  + sleep 5
07-20 16:49:24.185  + '[' 2 -ge 3 ']'
07-20 16:49:24.185  + hypershift_create_s3_bucket aos-hypershift-ci-oidc-5811 us-east-1
07-20 16:49:24.185  + local bucket_name=aos-hypershift-ci-oidc-5811
07-20 16:49:24.185  + local bucket_region=us-east-1
07-20 16:49:24.185  + aws s3api head-bucket --bucket aos-hypershift-ci-oidc-5811
07-20 16:49:24.743  /usr/local/lib/python3.6/site-packages/OpenSSL/_util.py:6: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography. The next release of cryptography will remove support for Python 3.6.
07-20 16:49:24.743    from cryptography.hazmat.bindings.openssl.binding import Binding
07-20 16:49:24.743  
07-20 16:49:24.743  An error occurred (404) when calling the HeadBucket operation: Not Found
07-20 16:49:24.743  + '[' 255 -ne 0 ']'
07-20 16:49:24.743  + aws s3api create-bucket --create-bucket-configuration LocationConstraint=us-east-1 --region us-east-1 --bucket aos-hypershift-ci-oidc-5811
07-20 16:49:25.301  /usr/local/lib/python3.6/site-packages/OpenSSL/_util.py:6: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography. The next release of cryptography will remove support for Python 3.6.
07-20 16:49:25.301    from cryptography.hazmat.bindings.openssl.binding import Binding
07-20 16:49:25.301  
07-20 16:49:25.301  An error occurred (InvalidLocationConstraint) when calling the CreateBucket operation: The specified location-constraint is not valid
```

**Comment by xiuwang** [comment]
@cdoern You are right, I can launch the hypershift hosted cluster. 
image-registry-ca cm in openshift-config-managed  keeps same content with image-registry-certificates cm in openshift-image-registry except the additionalTrustedCA

**Pull request merged**